### PR TITLE
Lang lib update

### PR DESCRIPTION
**Pull request by hasithaa**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/lang.array/src/main/ballerina/src/lang.array/internal.bal [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public function __init(Type[] m) {
+        self.m = m;
+    }
+
+    # Return next member or nil if end of iteration is reached.
+    # + return - iterator result
+    public function next() returns record {|
+        Type value;
+    |}? = external;
+};
```
Comment (MaryamZi) [inline langlib/lang.array/src/main/ballerina/src/lang.array/internal.bal]: Missing a new line.

File: langlib/lang.future/src/main/ballerina/src/lang.future/future.bal [hunk 2]
```diff
@@ -14,7 +14,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-# Cancels the given future.
-#
-# + f - The future to test
-public function cancel(future<any> f) = external;
+# Requests cancellation of a future.
+# + f - the future to be cancelled
+# This sets the cancellation flag in the strand corresponding to `f`.
```
Comment (MaryamZi) [inline langlib/lang.future/src/main/ballerina/src/lang.future/future.bal]: Shouldn't this bit come before the documentation for the parameter?

File: langlib/lang.map/src/main/ballerina/src/lang.map/internal.bal [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public function __init(map<Type> m) {
+        self.m = m;
+    }
+
+    # Return the next member in map iterator, nil if end of iterator is reached.
+    # + return - iterator result
+    public function next() returns record {|
+        Type value;
+    |}? = external;
+};
```
Comment (MaryamZi) [inline langlib/lang.map/src/main/ballerina/src/lang.map/internal.bal]: Missing a new line.

File: langlib/lang.table/src/main/ballerina/src/lang.table/internal.bal [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    private table <anydata|error> m;
+
+    public function __init(table<anydata|error> m) {
+        self.m = m;
+    }
+
+    public function next() returns record {|
+        RowType value;
+    |}? = external;
+};
```
Comment (MaryamZi) [inline langlib/lang.table/src/main/ballerina/src/lang.table/internal.bal]: Missing a new line.

File: langlib/lang.future/src/main/ballerina/src/lang.future/future.bal (hunk 2 above)
Comment (hasithaa) [reply langlib/lang.future/src/main/ballerina/src/lang.future/future.bal]: Apparently this need to be fixed in the spec. Fixed it locally.

**Pull request merged**